server/service/src/sync/translations/user_store_permissions.rs

```diff
@@ -0,0 +1,185 @@
+            for permission in existing_permissions {
+                integration_operations.push(IntegrationOperation::delete(UserPermissionRowDelete(
+                    permission.id,
+                )))
+            }
+            return Ok(PullTranslateResult::IntegrationOperations(
+                integration_operations,
+            ));
+        }
+
+        integration_operations.push(IntegrationOperation::upsert(user_store_join_row));
+
+        let new_permissions = map_api_permissions(permissions);
+        let mut new_permission_set = permissions_to_domain(new_permissions);
+        new_permission_set.insert(PermissionType::StoreAccess);
+
+        //TODO OG, when you add a user to a group it needs to requeue the user's own user_store records, not the group's. Just save them. Or save the groups user_store records, the trigger will do the members'
+        //TODO OG, test group sync permissions
+
+        // Similar to user_store_join, the login functionality will drop literally all the user's permissions and recreate them with new PKs.
+        // If the sync record turns the permission on and it exists, do nothing, else upsert a new record.
+        // If the sync record turns the permission off, delete the corresponding record.
+        // We cannot drop them all and insert again as login does as sync operations execute all deletes after all inserts, so we'd wipe out our permissions
+        let mut existing_permissions: HashMap<PermissionType, UserPermissionRow> =
+            existing_permissions
+                .into_iter()
+                .map(|p| (p.permission.clone(), p))
+                .collect();
+
+        for permission in new_permission_set {
+            if existing_permissions.remove(&permission).is_none() {
+                integration_operations.push(IntegrationOperation::upsert(UserPermissionRow {
+                    id: uuid(),
+                    user_id: user_id.clone(),
+                    store_id: Some(store_id.clone()),
+                    permission: permission.clone(),
+                    context_id: None,
+                }));
+            }
+        }
+
+        // Some prefs come from om_user_permission! They should be preserved so the `user_permission` translator can handle them
+        existing_permissions.remove(&PermissionType::DocumentQuery);
+        existing_permissions.remove(&PermissionType::DocumentMutate);
+
+        for (_, row) in existing_permissions {
+            integration_operations.push(IntegrationOperation::delete(UserPermissionRowDelete(
+                row.id,
+            )))
+        }
+
+        Ok(PullTranslateResult::IntegrationOperations(
+            integration_operations,
+        ))
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use repository::{mock::MockDataInserts, test_db::setup_all};
+
+    #[actix_rt::test]
+    async fn test_user_permission_translation() {
+        use crate::sync::test::test_data::user_permission as test_data;
```

Comment:
This test is copy pasta without modification - I need to write appropriate tests still!

(user_permission test data is for legacy `om_user_permission` table which controls a couple document permissions)